### PR TITLE
fix: explicitely check if data-hover validates to false

### PR DESCRIPTION
**Pull request by DanielRuf**
Reference: https://github.com/zurb/foundation-sites/issues/10685

### CODE REVIEW

**Comment by DanielRuf** [comment]
Todo: strict check for more attributes.

File: js/foundation.dropdown.js [hunk 1]
```diff
@@ -150,14 +150,14 @@ class Dropdown extends Positionable {
...
-        } else if (hasTouch && _this.options.hover && _this.$element.hasClass('is-open') === false) {
+        } else if (hasTouch && _this.options.hover !== false && _this.$element.hasClass('is-open') === false) {
           // if forceFollow true and hover option true, only prevent default action on 1st click
           // on 2nd click (dropown opened) the default action (e.g. follow a href) gets executed
           e.preventDefault();
         }
     });
 
-    if(this.options.hover){
+    if(this.options.hover !== false){
```
Comment (ncoden) [inline js/foundation.dropdown.js]: With `<div data-dropdown data-hover></div>`, `this.options.hover` is equal to an empty string here. So I wonder, do we need to:
- consider all non-false values as positives, which is a classic HTML approach
- ignore all invalid properties (expect a boolean but got a string), which is more sustainable approach.

The problem here is that we work with both HTML and Js values, and that in HTML `attribute` equals to `attribute=""`. Also, HTML boolean attibutes as considered as `true` as soon as they are set, even with a `"false"` as value...

With the current PR, `Foundation.Dropdown.defaults.hover = null` will make it considered as `true` even if its default is `false`.

**Comment by DanielRuf** [comment]
So check better for strict true or any type of string?

File: js/foundation.dropdown.js (hunk 1 above)
Comment (DanielRuf) [reply js/foundation.dropdown.js]: So check better for strict true or any type of string?

**Comment by ncoden** [comment]
> So check better for strict true or any type of string?

I don't know what solution to choose.

So I wonder, do we need to:
* consider all non-false values as positives
  ```
  if (this.options.hover !== false) ...
  ```
* ignore invalid properties (and use defaults instead)
  ```
  const validOptions = filterOptions({  }, Dropdown.defaults, this.$element.data(), options);
  this.options = filterOptions({ hover: Boolean, ... }, Dropdown.defaults, this.$element.data(), options);
  ```
* ignore invalid properties, but transform to true for Boolean
  (same code as above)

**Comment by DanielRuf** [comment]
Did some changes to cover empty attr values and boolean true.

**Comment by ncoden** [comment]
It's not only about this component. It's about how the whole framework should deal with Boolean/Empty attributes. We cannot make a manual check in each component for `true || ''`.

**Comment by DanielRuf** [comment]
Right. Another approach should be better (the internal plugin system).

Where and how could we implement this in the best way?

**Comment by DanielRuf** [comment]
@colin-marshall what do you think?

**Comment by ncoden** [comment]
The simplest would be to use a small utility like
```js
// foundation.util.core.js
function boolAttr(value) {
  return value === true || (typeof value === 'string' && !value.length);
}
```
```js
// foundation.dropdown.js
if (boolAttr(this.option.hover)) ...
```

But it would be a bit dirty.

An other solution is (like given in a previous comment) a filter when initializing options. But it would be the part of a way bigger architecture change to efficiently control typing (typescript would not be a bad idea btw).

**Comment by DanielRuf** [comment]
Should we plan this for v7 or how do we proceed with this PR?

**Comment by ncoden** [comment]
I think it's not too much work to go with a little helpers in all components like:

```js
    _init() {
        this.options = $.extend({}, Dropdown.defaults, this.$element.data(), options);
        this.options.hover = boolAttr(this.options.hover, 'hover');
    }
```

Or 

```js
    _init() {
        this.options = $.extend({}, Dropdown.defaults, this.$element.data(), options);
        this.options = parseBooleanAttributes(this.options, ['hover', ...]);
    }
```

**Comment by DanielRuf** [comment]
What should we choose? @colin-marshall @kball @Owlbertz @brettsmason

**Comment by ncoden** [comment]
@DanielRuf Personally I would go for the 3rd solution.

I can expect this to be `true`:
```html
<div data-dropdown data-hover>...<div>
```

But not this:
```js
Foundation.Dropdown.defaults.hover = '';
```

**Comment by DanielRuf** [comment]
@ncoden I guess you may know best how the helper should look and work like. Do you want to and can you take care of this?

**Comment by ncoden** [comment]
I'll take care of this, probably Saturday.

**Comment by DanielRuf** [comment]
I guess we have to rebase here. On which branch should I rebase?

**Comment by ncoden** [comment]
`develop`. But still, it's not a priority (v6.6.0).

**Pull request closed without merging**